client/src/app/shared/components/data-table/data-source/data-source.ts

```diff
@@ -0,0 +1,6 @@
+import {PagedData} from '../../../models/page';
+import {Observable} from 'rxjs';
+
+export interface DataSource<T> {
+  fetchData(options?: any): Observable<PagedData<T>>;
```

Comment:
Why are you using a method named `fetchData` instead of the documented `connect()`. I think Material would automatically call the DataSource's `connect()` method so it wouldn't have to be called in `DataTableComponent.fetchData`

Reply:
but this is not a angular material data table, it's using ngx-datatable, we could name it similarly but fetchData seems more descriptive to me than connect.

Reply:
actually, i haven't thought of checking if ngx-datatable supports datasource lol let me check now ...

Reply:
Ah I see! My bad! I just assumed it was material

`fetchData` implies a one-time, promise-based, call to get data, whereas `connect()` implies connecting to an updating data-source and reflecting the latest data from that source. So, unless I'm mistaken and this data is only fetched once, connect would be better. [This page](https://medium.com/angular-in-depth/angular-material-pagination-datasource-73080d3457fe) has a nice description of data sources using material.

Reply:
Actaully, I guess it is only fetched once for now so fetchData is not so bad but if we introduce pagination/filtering etc. at some point connect would be better IMO

Reply:
yeah, i'm looking at same article now! I should have reused the DataSource in `@angular/cdk/collections` and i could actually have both `connect` and fetchData/fetch, currently fetchData is also being called when page number changed.
it's like connect and fetch in 1 (as in that article).

Reply:
Hmm okay, i like to follow mainstream, i'll apply those changes when i get a chance again. Could I push the changes for now? :)

Reply:
Yeah sure go for it!